stdlib/public/Synchronization/Atomics/AtomicFloats.swift

```diff
@@ -118,7 +118,7 @@ extension Float: AtomicRepresentable {
 // Double AtomicRepresentable conformance
 //===----------------------------------------------------------------------===//
 
-#if (_pointerBitWidth(_32) && _hasAtomicBitWidth(_64)) || _pointerBitWidth(_64)
+#if ((_pointerBitWidth(_32) || _pointerBitWidth(_16)) && _hasAtomicBitWidth(_64)) || _pointerBitWidth(_64)
```

Comment:
I wonder whether this is actually meaningful on AVR? There is no floating point HW support, correct? If floating point is all software handled, is Float and Double actually "atomic representable"?

Reply:
Without this we get compilation failures in other parts of the standard library that expect to have AtomicRepresentation. I need to revert this change to find the compilation error for you.

Reply:
in answer, AVR has some very basic operations like FMUL on most but not all platforms (floating point 8 bit multiplication). No hardware floating point division. Imagine a Time Machine to CPUs from the early 90s. :)

Reply:
Hmm, now I wonder if we could just for the sake of keeping the size of this PR sane, postpone the porting of the Synchronization library (by not building it for AVR with the same CMake logic you use in the Unicode library). Would you mind doing that and then we can follow up on this after we get the stdlib to build? Thanks :)

Reply:
I tried reverting this and it seems fine, the standard library still compiles. I'll ditch it.